contrib/fd-plugins/bareos_tasks/BareosFdTaskClass.py

```diff
@@ -155,7 +155,12 @@ def execute_command(self, command):
     def pool(self):
         if self.use_stderr:
             try:
-                self.stderr_buffer.write(self.process.stderr.read())
+                while True:
+                    stderrtext=self.process.stderr.read()
+                    if stderrtext:
+                        self.stderr_buffer.write(stderrtext)
+                    else:
+                        break
```

Comment:
Can you describe, when the previous version failed?

Reply:
The old code failed on some backups. The error we had was that it was operating on a nonetype. And effectively the "stderrtext" would be empty in certain cases.